soak/README.md

```diff
@@ -0,0 +1,202 @@
+* RUN `export CONTROLLER_CHART_RELEASE_NAME=<controller-chart-release-name>` Ex: soak-test
+* RUN `helm chart pull $CONTROLLER_CHART_URL`
+* RUN `helm chart install $CONTROLLER_CHART_RELEASE_NAME $CONTROLLER_CHART_URL` 
+
+b) Use the chart from source repository
+
+* RUN `export SERVICE_NAME=<aws-service-name>` Ex: 'apigatewayv2'
+* checkout `<aws-service-name>-controller` github repository
+* RUN `cd helm`
+* RUN `export CONTROLLER_CHART_RELEASE_NAME=<controller-chart-release-name>` Ex: soak-test
+* RUN `helm install $CONTROLLER_CHART_RELEASE_NAME .` Note: Set custom values during helm install
+
+NOTE
+* Use "--set serviceAccount.Annotations.eks.amazonaws.com/role-arn=<IAM_ROLE_ARN_FOR_IRSA>" during helm install.
+* We highly recommend using IRSA for providing AWS permissions to your service controller. See [installation docs](https://aws-controllers-k8s.github.io/community/user-docs/install/)
+ for how to use IRSA for ACK service controller pod.
+
+After Step1 you will have ack-service-controller running in your cluster, exposing ack metrics through a K8s service endpoint.
+Service name will be `$CONTROLLER_CHART_RELEASE_NAME-ack-$SERVICE_NAME-controller` Ex: "soak-test-ack-apigatewayv2-controller"
+
+
+### STEP 2 (Install "kube-prometheus" chart for monitoring the soak test behavior)
+
+NOTE: You can skip this step if you have already installed 'Prometheus', 'Grafana' or other monitoring mechanism in your K8s cluster
+where the soak tests are running.
+
+The easiest way to install 'Prometheus', 'Grafana', 'Node Exporter' and 'Kube State Metrics' is through the official helm chart.
+Follow the commands below to install this helm chart.
+
+* RUN `helm repo add prometheus-community https://prometheus-community.github.io/helm-charts`
+* RUN `export PROM_CHART_RELEASE_NAME=<kube-prometheus-chart-release-name>` Ex: prom
+* RUN `helm install $PROM_CHART_RELEASE_NAME prometheus-community/kube-prometheus-stack`
+* Prometheus and Grafana services will be started as ClusterIP services.
+* To access the console you can use the following instructions
+    * PROMETHEUS
+       * Run `kubectl port-forward service/$PROM_CHART_RELEASE_NAME-kube-prometheus-stack-prometheus 9090:9090`
+       * Now you can view Prometheus metrics at `http://127.0.0.1:9090/`
+    * GRAFANA
+        * Run `kubectl port-forward service/$PROM_CHART_RELEASE_NAME-grafana 3000:80`
+        * Now you can view Grafana dashboards at `http://127.0.0.1:3000/`
+        * Default Username for Grafana login is `admin` and default password is `prom-operator`
+        * Note: You can also see these username and password in secret `$PROM_CHART_RELEASE_NAME-grafana`
+* After logging into Grafana dashboard, to view resource metrics of controller-pod,
+    * Go to `http://127.0.0.1:3000/dashboards`
+    * Under `Manage` tab find `Kubernetes/Compute Resources/Pod`
+    * And load the metrics for service-controller pod from the pod selection.
```

Comment:
We are going to need to automate this process for the soak tests? Is it possible to just provide the script to automate it here?